cirq/ops/thermal_relaxation_channel.py

```diff
@@ -0,0 +1,267 @@
+    """Dampen qubits at varying rates through non ideal disipation.
+
+    This channel models the effect of energy dissipation into the environment
+    as well as the environment depositing energy into the system at varying
+    rates.
+    """
+
+    def __init__(self, p_exchange: float, p_relaxation: float,
+                 p_dephasing: float) -> None:
+        r"""The thermal relaxation channel.
+
+        Construct a channel to model energy dissipation into the environment
+        as well as the environment depositing energy into the system. The
+        probabilities with which the energy exchange occur are given by `gamma`
+        and `beta` with the probability of the environment being not excited is
+        given by `p`.
+
+        The stationary state of this channel is the diagonal density matrix
+        with probability `p` of being |0⟩ and probability `1-p` of being |1⟩.
+
+        Recall the kraus opeartors for the `GeneralizedAmplitudeDampingChannel`:
+
+            $$
+            \begin{aligned}
+            M_0 =& \sqrt{p_exchange} \begin{bmatrix}
+                                1 & 0  \\
+                                0 & \sqrt{1 - p_relaxation}
+                            \end{bmatrix}
+            \\
+            M_1 =& \sqrt{p_exchange} \begin{bmatrix}
+                                0 & \sqrt{p_relaxation} \\
+                                0 & 0
+                           \end{bmatrix}
+            \\
+            M_2 =& \sqrt{1-p_exchange} \begin{bmatrix}
+                                \sqrt{1-p_relaxation} & 0 \\
+                                 0 & 1
+                              \end{bmatrix}
+            \\
+            M_3 =& \sqrt{1-p_exchange} \begin{bmatrix}
+                                 0 & 0 \\
+                                 \sqrt{p_relaxation} & 0
+                             \end{bmatrix}
+            \end{aligned}
+            $$
+
+        This has a choi representation of:
+
+            $$
+             \begin{aligned}
+                CHO =&  \begin{bmatrix}
+                            1 - (1-p_exchange) p_relaxation & 0 & 0 &
+                                \sqrt{1 - p_relaxation} \\
+                            0 & (1-p_exchange) p_relaxation 0 & 0 \\
+                            0 & 0 & p_exchange p_relaxation & 0 \\
+                            \sqrt{1 - p_relaxation} & 0 & 0 &
+                                1 - p_exchange p_relaxation
+                        \end{bmatrix}
+             \end{aligned}
+            $$
+
+        Where each 2x2 block can be thought of as how the channel acts on each
+        basis element in the space of density matrices.
+
+        For the `ThermalRelaxationChannel` channel we simply substitute
+        `p_dephasing` in for `p_relaxation` on the off diagonal elements of the
+        above choi matrix and then evolve the channel using the kraus operators
+        found by sqrt factorization of the choi matrix.
```

Comment:
This comment makes the generalized amplitude damping channel very prominent and buries the definition of the thermal relaxation channel at the very end.

We should first define the thermal relaxation channel and then point out how it relates to the generalized amplitude damping channel. If the Kraus operators for the thermal relaxation channel are indeed to complicated to write down (see another comment for a hint of what can be exploited to help simplify them), then let's define the thermal relaxation channel in terms of its Choi matrix.

I don't think we should be displaying Kraus operators for the generalized amplitude damping channel here at all.

Reply:
I've removed the kraus ops for the GAD channel and updated the docstring, let me know what you think.